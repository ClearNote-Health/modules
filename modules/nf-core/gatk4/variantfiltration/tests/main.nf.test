nextflow_process {

    name "Test Process GATK4_VARIANTFILTRATION"
    script "../main.nf"
    process "GATK4_VARIANTFILTRATION"

    tag "modules"
    tag "modules_nfcore"
    tag "gatk4"
    tag "gatk4/variantfiltration"

    test("test_gatk4_variantfiltration_vcf_input") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.idx', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
                ]
                input[3] = [
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.dict', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() },
                { assert snapshot(file(process.out.vcf.get(0).get(1)).name).match("gatk_hc_vcf_bam_gz_input") },
                { assert snapshot(file(process.out.tbi.get(0).get(1)).name).match("gatk_hc_vcf_tbi_bam_gz_input") },
            )
        }

    }

    test("test_gatk4_variantfiltration_gz_input") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz.tbi', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
                ]
                input[3] = [
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.dict', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() },
                { assert snapshot(file(process.out.vcf.get(0).get(1)).name).match("gatk_hc_vcf_bam_input") },
                { assert snapshot(file(process.out.tbi.get(0).get(1)).name).match("gatk_hc_vcf_tbi_bam_input") },
            )
        }

    }

}

