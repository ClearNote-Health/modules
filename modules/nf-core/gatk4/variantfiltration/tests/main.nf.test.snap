{
    "test_gatk4_variantfiltration_gz_input": {
        "content": [
            [
                "versions.yml:md5,96943659275ba62de1f0d283a2f6e97b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-22T22:25:07.54709"
    },
    "gatk_hc_vcf_bam_input": {
        "content": [
            "test.vcf.gz"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-22T22:25:07.553307"
    },
    "gatk_hc_vcf_bam_gz_input": {
        "content": [
            "test.vcf.gz"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-22T22:24:27.86075"
    },
    "gatk_hc_vcf_tbi_bam_input": {
        "content": [
            "test.vcf.gz.tbi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-22T22:25:07.559474"
    },
    "test_gatk4_variantfiltration_vcf_input": {
        "content": [
            [
                "versions.yml:md5,96943659275ba62de1f0d283a2f6e97b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-22T22:24:27.841897"
    },
    "gatk_hc_vcf_tbi_bam_gz_input": {
        "content": [
            "test.vcf.gz.tbi"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-05-22T22:24:27.865577"
    }
}